src/main/java/com/devonfw/ide/sonarqube/common/impl/DevonfwJavaProfile.java

```diff
@@ -0,0 +1,31 @@
+package com.devonfw.ide.sonarqube.common.impl;
+
+import org.sonar.api.server.profile.BuiltInQualityProfilesDefinition;
+import org.sonar.api.utils.AnnotationUtils;
+import org.sonar.plugins.java.api.JavaCheck;
+import org.sonarsource.api.sonarlint.SonarLintSide;
+
+/**
+ * @author lniazman
+ *
+ */
+@SonarLintSide
+public class DevonfwJavaProfile implements BuiltInQualityProfilesDefinition {
+
+  @Override
+  public void define(Context context) {
+
+    NewBuiltInQualityProfile devonfwJava = context.createBuiltInQualityProfile("Devonfw Java", "java");
+    org.sonar.check.Rule ruleAnnotation;
+
+    for (Class<? extends JavaCheck> check : DevonSonarRegistrar.checkClasses()) {
+
+      ruleAnnotation = AnnotationUtils.getAnnotation(check, org.sonar.check.Rule.class);
+      devonfwJava.activateRule(DevonSonarDefinition.REPOSITORY_KEY, ruleAnnotation.key());
+    }
+
+    devonfwJava.setDefault(true);
```

Comment:
When is this executed? Every time the SonarQube server starts up? We can not make this the default profile all the time or users will get totally annoyed. We should only do that the first time the quality profile is created. In case it already exists, the default attribute should not be changed anymore.

Reply:
How about putting a note into the readme, that the quality profile has to be set as default or associated with a project manually in the SQ server GUI? I think it makes sense that the default Sonar profile stays as default.